From c0671f46bd9ac1beb45d7abc66fbfd3311fd7bbc Mon Sep 17 00:00:00 2001
From: Yaroslav Halchenko <debian@onerussian.com>
Date: Thu, 13 Aug 2020 09:55:33 -0400
Subject: [PATCH] BF(?): serve_path_via_http,HTTPPath - use "localhost" not
 hardcoded IPv4

In hope that it could potentially work on IPv6 only setups
---
 datalad/tests/utils.py | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

--- a/datalad/tests/utils.py
+++ b/datalad/tests/utils.py
@@ -634,14 +634,9 @@ class HTTPPath(object):
     def start(self):
         """Start serving `path` via HTTP.
         """
-        # There is a problem with Haskell on wheezy trying to
-        # fetch via IPv6 whenever there is a ::1 localhost entry in
-        # /etc/hosts.  Apparently fixing that docker image reliably
-        # is not that straightforward, although see
-        # http://jasonincode.com/customizing-hosts-file-in-docker/
-        # so we just force to use 127.0.0.1 while on wheezy
-        #hostname = '127.0.0.1' if on_debian_wheezy else 'localhost'
-        hostname = '127.0.0.1'
+        # Generally avoid IPv4 addresses, use names, so testing could be done
+        # also on IPv6 only setups (see https://bugs.debian.org/963856)
+        hostname = 'localhost'
 
         queue = multiprocessing.Queue()
         self._mproc = multiprocessing.Process(
